server/keyshare/keyshareserver/server.go

```diff
@@ -240,6 +298,71 @@ func (s *Server) generateCommitments(user *User, authorization string, keys []ir
 	return &irma.ProofPCommitmentMap{Commitments: mappedCommitments}, nil
 }
 
+// /api/v2/prove/getCommitments
+func (s *Server) handleCommitmentsV2(w http.ResponseWriter, r *http.Request) {
+	// Fetch from context
+	user := r.Context().Value("user").(*User)
+	authorization := r.Context().Value("authorization").(string)
+
+	// Read keys
+	var req irma.GetCommitmentsRequest
+	if err := server.ParseBody(r, &req); err != nil {
+		server.WriteError(w, server.ErrorInvalidRequest, err.Error())
+		return
+	}
+	if len(req.Keys) == 0 {
+		s.conf.Logger.Info("Malformed request: no keys for commitment specified")
+		server.WriteError(w, server.ErrorInvalidRequest, "No key specified")
+		return
+	}
+
+	commitments, err := s.generateCommitmentsV2(user, authorization, req)
+	// TODO: can ErrInvalidChallenge be removed?
+	if err != nil && (err == keysharecore.ErrInvalidChallenge || err == keysharecore.ErrInvalidJWT) {
+		server.WriteError(w, server.ErrorInvalidRequest, err.Error())
+		return
+	}
+	if err != nil {
+		// already logged
+		server.WriteError(w, server.ErrorInternal, err.Error())
+		return
+	}
+
+	server.WriteJson(w, commitments)
+}
+
+func (s *Server) generateCommitmentsV2(user *User, authorization string, req irma.GetCommitmentsRequest) (*irma.ProofPCommitmentMapV2, error) {
+	// Generate commitments
+	commitments, commitID, err := s.core.GenerateCommitments(user.Secrets, authorization, req.Keys)
+	if err != nil {
+		s.conf.Logger.WithField("error", err).Warn("Could not generate commitments for request")
+		return nil, err
+	}
+
+	// Prepare output message format
+	// TODO: move logic to gabi?
+	mappedCommitments := map[irma.PublicKeyIdentifier]*big.Int{}
+	for i, keyID := range req.Keys {
+		mappedCommitments[keyID] = commitments[i].Pcommit
+	}
```

Comment:
What's the aim of this TODO?

Reply:
Due to legacy we get more back from gabi than we need and need to do an extra mapping here. I put down a TODO because I'd like to evaluate again later when the rest is implemented if we still want to move some things from gabi to irmago or the other way round.

In this case gabi could also return a `map[irma.PublicKeyIdentifier]*big.Int{}` directly.